homeassistant/components/solax/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Config flow for solax integration."""
+import logging
+
+from solax import real_time_api
+from solax.inverter import DiscoveryError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+DEFAULT_PASSWORD = ""
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+    }
+)
+
+
+async def validate_api(data):
+    """Validate the credentials."""
+
+    api = await real_time_api(
+        data[CONF_IP_ADDRESS], data[CONF_PORT], data[CONF_PASSWORD]
+    )
+    resp = await api.get_data()
+    return resp.serial_number
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Solax."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
```

Comment:
Typing is not required, but would be nice to add it here.

Reply:
You mean the user_input variable right? I removed the VERSION as it was not being used.